tests/components/iskra/test_config_flow.py

```diff
@@ -0,0 +1,241 @@
+"""Tests for the Iskra config flow."""
+
+from unittest.mock import patch
+
+from pyiskra.exceptions import (
+    DeviceConnectionError,
+    DeviceTimeoutError,
+    InvalidResponseCode,
+    NotAuthorised,
+)
+import pytest
+
+from homeassistant.config_entries import SOURCE_USER
+from homeassistant.const import (
+    CONF_ADDRESS,
+    CONF_HOST,
+    CONF_PASSWORD,
+    CONF_PORT,
+    CONF_PROTOCOL,
+    CONF_USERNAME,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResultType
+
+from tests.common import MockConfigEntry
+
+SG_MODEL = "SG-W1"
+PQ_MODEL = "MC784"
+SERIAL = "XXXXXXX"
+HOST = "192.1.0.1"
+MODBUS_PORT = 10001
+MODBUS_ADDRESS = 33
+
+
+class MockBasicInfo:
+    """Mock BasicInfo class."""
+
+    def __init__(self, model):
+        """Initialize the mock class."""
+        self.serial = SERIAL
+        self.model = model
+        self.description = "Iskra mock device"
+        self.location = "imagination"
+        self.sw_ver = "1.0.0"
+
+
+@pytest.fixture
+def mock_pyiskra_rest():
+    """Mock Iskra API authenticate with Rest API protocol."""
+
+    with patch(
+        "pyiskra.adapters.RestAPI.RestAPI.get_basic_info",
+        return_value=MockBasicInfo(model=SG_MODEL),
+    ) as basic_info_mock:
+        yield basic_info_mock
+
+
+@pytest.fixture
+def mock_pyiskra_modbus():
+    """Mock Iskra API authenticate with Rest API protocol."""
+
+    with patch(
+        "pyiskra.adapters.Modbus.Modbus.get_basic_info",
+        return_value=MockBasicInfo(model=PQ_MODEL),
+    ) as basic_info_mock:
+        yield basic_info_mock
+
+
+# Test step_user with Rest API protocol
+async def test_user_rest_no_auth(hass: HomeAssistant, mock_pyiskra_rest) -> None:
+    """Test the user flow with Rest API protocol."""
+
+    result = await hass.config_entries.flow.async_init(
+        "iskra",
+        context={"source": SOURCE_USER},
+    )
+
+    assert result["type"] is FlowResultType.FORM
+    assert result["step_id"] == "user"
+
+    # Test no authentication required
+    result = await hass.config_entries.flow.async_init(
+        "iskra",
+        context={"source": SOURCE_USER},
+        data={CONF_HOST: HOST, CONF_PROTOCOL: "rest_api"},
+    )
+
+    # Test successful Rest API configuration
+    assert result["type"] is FlowResultType.CREATE_ENTRY
+    assert result["title"] == f"{SG_MODEL} {SERIAL}"
+    assert result["data"][CONF_HOST] == HOST
```

Comment:
Let's assert the full data dict, and the unique_id

Reply:
i did assert full data dict now but don't know how to do so for unique_id,

Reply:
What I originally meant was
```suggestion
    assert result["data"] = {
        CONF_HOST: HOST,
        ...
    }
```

Reply:
```py
assert result["result"].unique_id == "..."
```